### PR TITLE
Add "-Wc,list(),offset,gonumber" compile options on z/OS

### PR DESCRIPTION
**Pull request by pshipton**
In order to create compiler pseudo assembly listing .lst files.

https://www.ibm.com/docs/en/zos/2.4.0?topic=options-list-nolist
https://www.ibm.com/docs/en/zos/2.4.0?topic=options-offset-nooffset
https://www.ibm.com/docs/en/zos/2.4.0?topic=options-gonumber-nogonumber

### CODE REVIEW

**Comment by pshipton** [comment]
btw, the debug image file is a tar.gz although there is https://github.com/eclipse-openj9/openj9/blob/master/buildenv/jenkins/common/build.groovy#L387. Checking with service the tar.gz is preferred over pax for the debug image.

File: runtime/cmake/platform/os/zos.cmake [hunk 1]
```diff
@@ -25,4 +25,7 @@ set(CMAKE_SHARED_LIBRARY_PREFIX "lib")
 
 set(CMAKE_IMPORT_LIBRARY_PREFIX "lib")
 
-list(APPEND OMR_PLATFORM_COMPILE_OPTIONS "\"-Wc,inline(auto,noreport,600,5000)\"")
+list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+    "\"-Wc,inline(auto,noreport,600,5000)\""
+    "\"-Wc,list(),offset,gonumber\""
```
[the remark this answers is not in the document]
Comment (pshipton) [reply runtime/cmake/platform/os/zos.cmake:30]: I'll give it a try.

[the remark this answers is not in the document]
Comment (pshipton) [reply runtime/cmake/platform/os/zos.cmake:30]: Without the brackets the listing file goes to the console.
job/Build_JDK11_s390x_zos_Personal/401

**Comment by keithc-ca** [comment]
I think we should be consistent and stick with `pax` on z/OS.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**